public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts

```diff
@@ -299,16 +314,28 @@ export default class AzureLogAnalyticsDatasource extends DataSourceWithBackend<
     return quotedValues.join(',');
   }
 
-  getDefaultOrFirstWorkspace() {
+  async getDefaultOrFirstSubscription(): Promise<string | undefined> {
+    if (this.defaultSubscriptionId) {
+      return this.defaultSubscriptionId;
+    }
+    return this.getSubscriptions().then((subs) => first(subs.map((s) => s.value)));
```

Comment:
Is there something specific that lodash first() gets us apart from just `arr[0]`?

This is an async function, so you could just do:

```typescript
const subscriptions = await this.getSubscriptions();
return subscriptions[0]?.value
```

maybe?

Reply:
Changed